### PR TITLE
refactor: cost-saving with AMD instances

### PR DESCRIPTION
**Pull request by csivanich**
## Why are these changes needed?

Improves cost of test execution by switching from Intel to AMD backed instances.

Saves ~10% on all `m5.*` instance types with similar baseline performance and full binary compatibility.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by DmitriGekhtman** [comment]
Looks like this modifies the example configurations in `python/ray/autoscaler/aws` too. These aren't used for tests, but I imagine users who are trying out Ray for the first time wouldn't mind the cost improvement.

Does anyone else mind this change in the example files?

**Review by krfricke — CHANGES_REQUESTED** [review]
The main thing I'm concerned about is instance availability and quota. If @thomasdesr can confirm we have enough quota for m5a.xlarge/2xlarge instances I'm happy to go ahead.

We need to run a suite of release tests on these instances to make sure they are still pasisng. I can kick these off once the quota is confirmed.

For the buildkite `pipeline.ml.yml` tests, they currently hang as our aws cloudformation stack does not have these instances defined. We are currently looking into changing the queues anyway, until then I'd suggest we revert that specific change.

**Comment by thomasdesr** [comment]
You should have access to go check the quotas in the test account, but in this case I believe they share the same quota family! So we should be good to go here.

**Comment by csivanich** [comment]
@DmitriGekhtman I'm actually curious about your thoughts on changing examples/demos in the codebase. I tried to leave _most_ of the docs/examples untouched since the behavior/perf should be roughly the same. I can certainly take another pass and update them too.

**Comment by csivanich** [comment]
@krfricke 

> For the buildkite pipeline.ml.yml tests, they currently hang as our aws cloudformation stack does not have these instances defined. We are currently looking into changing the queues anyway, until then I'd suggest we revert that specific change.

For my knowledge can you point me to where the CF stack is defined? I can pull that file out of the changes in the meantime.

**Comment by DmitriGekhtman** [comment]
> @DmitriGekhtman I'm actually curious about your thoughts on changing examples/demos in the codebase. I tried to leave _most_ of the docs/examples untouched since the behavior/perf should be roughly the same. I can certainly take another pass and update them too.

I'm leaning conservative -- it's preferable if the examples are untouched, since at the moment we don't have any tests for them. 
...These examples have a tendency to break spontaneously even without us changing the configs.

**Comment by krfricke** [comment]
@csivanich it's here: https://github.com/ray-project/buildkite-ci-stack/blob/master/ci-stack-module/main.tf#L59

Let's keep the changes just for the release tests then, kick them off, and see if everything passes.

File: release/microbenchmark/tpl_64.yaml [hunk 1]
```diff
@@ -5,11 +5,11 @@ max_workers: 0
 
 head_node_type:
     name: head_node
-    instance_type: m5.16xlarge
+    instance_type: m5a.16xlarge
```
Comment (wuisawesome) [inline release/microbenchmark/tpl_64.yaml:8]: Probably don't want to change this (consistent benchmarks are pretty useful)

**Review by wuisawesome — CHANGES_REQUESTED** [review]
May not want to change the microbenchmarks. Not sure if the other ones should be changed or not.

**Pull request closed without merging**